src/main/scala/devices/debug/Debug.scala

```diff
@@ -324,16 +425,20 @@ class TLDebugModuleOuter(device: Device)(implicit p: Parameters) extends LazyMod
       when (~dmactive) {
         debugIntNxt(component) := false.B
       }. otherwise {
-        when (DMCONTROLWrEn && DMCONTROLWrData.hartsello === component.U) {
+        when (DMCONTROLWrEn && ((DMCONTROLWrData.hartsello === component.U) || (supportHartArray.B && DMCONTROLWrData.hasel && hamask(component)))) {
```

Comment:
It looks like you pulled out the `supportHartArray.B ... ` clause elsewhere, at least I don't see it anymore. Is there a reason why this one is still here?